Include/Python.h

```diff
@@ -55,6 +55,10 @@
 #  include <intrin.h>             // __readgsqword()
 #endif
 
+#if defined(Py_GIL_DISABLED) && defined(MS_WINDOWS) && defined(__GNUC__)
```

Comment:
Is `__GNUC__` the right test here? I thought that was cross-platform and represents the language version, rather than the compiler?

`MS_WINDOWS` is only going to be defined when building CPython using MSBuild or including headers that were built using MSBuild. So a Cygwin build (based around the makefile) won't have it set.

I can't find any documentation about the `intrin.h` header for anything except MSVC, and a few hints that Clang has a copy/compatibility layer. Which distributions include this header, and how can we identify them at build time?

Reply:
Can change it for __MINGW32__ if you prefer? It is included in mingw

Reply:
Also I guess if using ``__MINGW__`` also checking for ``MS_WINDOWS`` is no longer needed. Would you agree?